wsmaster/che-core-api-logger/src/main/java/org/eclipse/che/api/logger/LoggerService.java

```diff
@@ -0,0 +1,136 @@
+/*
+ * Copyright (c) 2012-2018 Red Hat, Inc.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *   Red Hat, Inc. - initial API and implementation
+ */
+package org.eclipse.che.api.logger;
+
+import static javax.ws.rs.core.MediaType.APPLICATION_JSON;
+import static org.eclipse.che.dto.server.DtoFactory.newDto;
+
+import ch.qos.logback.classic.Level;
+import ch.qos.logback.classic.Logger;
+import ch.qos.logback.classic.LoggerContext;
+import io.swagger.annotations.Api;
+import io.swagger.annotations.ApiOperation;
+import io.swagger.annotations.ApiParam;
+import io.swagger.annotations.ApiResponse;
+import io.swagger.annotations.ApiResponses;
+import java.util.List;
+import java.util.stream.Collectors;
+import javax.ws.rs.Consumes;
+import javax.ws.rs.GET;
+import javax.ws.rs.POST;
+import javax.ws.rs.PUT;
+import javax.ws.rs.Path;
+import javax.ws.rs.PathParam;
+import javax.ws.rs.Produces;
+import org.eclipse.che.api.core.NotFoundException;
+import org.eclipse.che.api.core.rest.Service;
+import org.eclipse.che.api.logger.shared.dto.LoggerDto;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Defines Logger REST API. It allows to manage the loggers (with log level) dynamically.
+ *
+ * @author Florent Benoit
+ */
+@Api(value = "/logger", description = "Logger REST API")
+@Path("/logger")
+public class LoggerService extends Service {
+
+  @GET
+  @Path("/{name}")
+  @Produces(APPLICATION_JSON)
+  @ApiOperation(value = "Get the logger level for the given logger")
+  @ApiResponses({
+    @ApiResponse(code = 200, message = "The response contains requested logger entity"),
+    @ApiResponse(code = 404, message = "The logger with specified name does not exist")
+  })
+  public LoggerDto getLoggerByName(@ApiParam(value = "logger name") @PathParam("name") String name)
+      throws NotFoundException {
+    return asDto(getLogger(name));
+  }
+
+  @GET
+  @Produces(APPLICATION_JSON)
+  @ApiOperation(
+    value = "Get loggers which are configured",
+    notes = "This operation can be performed only by authorized user",
+    response = LoggerDto.class,
+    responseContainer = "List"
+  )
+  @ApiResponses({
+    @ApiResponse(code = 200, message = "The loggers successfully fetched"),
+  })
+  public List<LoggerDto> getLoggers() {
```

Comment:
shall we add pagination here? with limit 30 by default

Reply:
We could yes but for now there is only 3 loggers
https://github.com/eclipse/che-lib/blob/master/che-tomcat8-slf4j-logback/src/assembly/conf/logback.xml#L47-L49

Reply:
added pagination